### PR TITLE
Floating Text

### PR DESCRIPTION
**Pull request by tomerasraf**
- Popup text can now can be called from IFloatingText.
- When player pick up a coin a text pop up appear with the amount text, and disappear after a certain duration.

### CODE REVIEW

File: Assets/Scripts/Features/Coins/CoinsVisual.cs [hunk 1]
```diff
@@ -62,6 +62,7 @@ void Update()
                         var adding = Feature.VolumeAmounts[coin.Volume - 1];
                         Feature.Wallet.AddToWallet(adding).Forget();
                         coin.Pick();
+                        Feature.FloatingText.PopText(coin.transform.position, coin.Volume.ToString(),2f);
```
Comment (ilia-gl) [inline Assets/Scripts/Features/Coins/CoinsVisual.cs:65]: The Volume of the Coin is the Size of it. 
The amount of Money is Feature.VolumeAmounts[coin.Volume - 1]

File: Assets/Scripts/Features/FloatingText/FloatingTextVisual.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        public void PopText(Vector3 position, string text, float duration)
+        {
+           popCoroutine = StartCoroutine(PopTextRoutine(position, text, duration));
+        }
+        
+        IEnumerator PopTextRoutine(Vector3 pos, string text, float duration)
+        {
+            var popText = Summoner.CreateAsset(floatingTextPrefab, transform);
+            var worldPos = UnityEngine.Camera.main.WorldToScreenPoint(pos);
```
Comment (ilia-gl) [inline Assets/Scripts/Features/FloatingText/FloatingTextVisual.cs:29]: Its better to access the Camera via the ICamera in the Feature.

Comment (ilia-gl) [reply Assets/Scripts/Features/FloatingText/FloatingTextVisual.cs:29]: Also its not
var worldPos
its
var screenPos

**Review by ilia-gl — APPROVED** [review]
Very good, this is another very reusable Feature for the many of our next games

File: Assets/Scripts/Features/Coins/CoinsVisual.cs (hunk 1 above)
Comment (tomerasraf) [reply Assets/Scripts/Features/Coins/CoinsVisual.cs:65]: Oh I see thanks

File: Assets/Scripts/Features/FloatingText/FloatingTextVisual.cs (hunk 2 above)
Comment (tomerasraf) [reply Assets/Scripts/Features/FloatingText/FloatingTextVisual.cs:29]: You are right sir

**Pull request merged**